AMM/project/scripts/swap-exact-output/src/main.sw

```diff
@@ -0,0 +1,59 @@
+script;
+
+use libraries::{AMM, Exchange};
+use std::block::height;
+
+enum InputError {
+    RouteTooShort: (),
+    SwapAmountsNotExact: (),
+}
+
+enum SwapError {
+    PairExchangeNotRegistered: (ContractId, ContractId),
+}
+
+fn main(
```

Comment:
This is trickier to do than the "exact input" case, as here you have no choice but to calculate all the swaps ahead of time. This can be done with helper functions on the AMM contract (see `get_amounts_in` on the [uniswap router contract](https://etherscan.io/address/0xf164fc0ec4e93095b804a4795bbe1e041497b92a#code), or basically what you already have in your test code)

This would enable the user to enter just the amount they want out at the end of the swap, as the script would calculate all the intermediate amounts.

Same as for the "exact input" case, the main function should take a deadline parameter, and a `maximum_amount_in` parameter to protect against excessive total slippage.